src/node/nodestate.h

```diff
@@ -205,6 +205,87 @@ namespace ccf
       sm.advance(State::initialized);
     }
 
+    std::vector<uint8_t> SerializeGenesis(
+      const CreateNew::In& args, std::vector<uint8_t>& quote)
+    {
+      jsonrpc::ProcedureCall<CreateNetworkNodeToNode::In> create_rpc;
+
+      create_rpc.id = 0;
+      create_rpc.method = ccf::MemberProcs::CREATE;
+
+      for (auto& cert : args.config.genesis.member_certs)
+      {
+        create_rpc.params.member_cert.push_back(cert);
+      }
+
+      create_rpc.params.gov_script = args.config.genesis.gov_script;
+      create_rpc.params.node_cert = node_cert;
+      create_rpc.params.network_cert = network.secrets->get_current().cert;
+      create_rpc.params.quote = quote;
+      create_rpc.params.code_digest =
+        std::vector<uint8_t>(std::begin(node_code_id), std::end(node_code_id));
+      create_rpc.params.node_info_network.host =
+        args.config.node_info_network.host;
+      create_rpc.params.node_info_network.pubhost =
+        args.config.node_info_network.pubhost;
+      create_rpc.params.node_info_network.nodeport =
+        args.config.node_info_network.nodeport;
+      create_rpc.params.node_info_network.rpcport =
+        args.config.node_info_network.rpcport;
+
+      nlohmann::json j = create_rpc;
+      auto contents = nlohmann::json::to_msgpack(j);
+
+      auto sig_contents = node_kp->sign(contents);
+
+      nlohmann::json sj;
+      sj["req"] = j;
+      sj["sig"] = sig_contents;
+
+      return jsonrpc::pack(sj, jsonrpc::Pack::Text);
+    }
+
+    void SendRequest(std::vector<uint8_t>& packed)
+    {
+      auto handler = this->rpc_map->find(ccf::ActorsType::members);
+      if (!handler.has_value())
+      {
+        LOG_INFO << "handler has no value" << std::endl;
+        return;
+      }
+      auto frontend = handler.value();
+
+      enclave::RPCContext ctx(
+        enclave::InvalidSessionId, node_cert, ccf::ActorsType::nodes);
+      ctx.is_create_request = true;
+
+#ifdef PBFT
+      frontend->process_pbft(ctx, packed);
+#else
+      frontend->process(ctx, packed);
+#endif
+    }
+
+    bool ApplyGenesisTx(const CreateNew::In& args, std::vector<uint8_t>& quote)
+    {
+      auto rpc = SerializeGenesis(args, quote);
+      // Become the primary and force replication.
+      consensus->force_become_primary();
+
+      SendRequest(rpc);
+
+      // Accept node connections for other nodes to join
```

Comment:
Could this until `return true;` be moved out of this function as it is not related to the genesis tx but the TLS connections + state of the node?